faunadb-scala/src/main/scala/faunadb/FaunaClient.scala

```diff
@@ -291,24 +298,24 @@ class FaunaClient private (connection: Connection) {
   def syncLastTxnTime(timestamp: Long): Unit =
     connection.syncLastTxnTime(timestamp)
 
-  private def parseResponseBody(responseBody: String)(implicit ec: ExecutionContext): Future[JsonNode] = {
-    def parse: Future[Option[JsonNode]] = Future(Option(json.readTree(responseBody)))
+  private def parseResponseBody(responseBody: String): Try[JsonNode] = {
+    def parse: Try[Option[JsonNode]] = Try(Option(json.readTree(responseBody)))
```

Comment:
We were spawning a new `Future` on purpose here (basically we want to do the parsing async). So, I'd say we revert those changes here. :)

Reply:
Do you mean performing the parsing on a different thread pool?

In any case I will revert this to make the diff. smaller :+1:

Reply:
https://github.com/fauna/faunadb-jvm/pull/281/commits/3f05ee13a89ca97847a93be4f9e8e2f66c67701c